modules/test/playwright/tests/portal-user-locale-options-web/userLocaleOptions.spec.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {expect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {dataApiHelpersTest} from '../../fixtures/dataApiHelpersTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {siteSettingsPagesTest} from '../../fixtures/siteSettingsPagesTest';
+import {usersAndOrganizationsPagesTest} from '../../fixtures/usersAndOrganizationsPagesTest';
+import getRandomString from '../../utils/getRandomString';
+
+export const test = mergeTests(
+	apiHelpersTest,
+	dataApiHelpersTest,
+	loginTest(),
+	siteSettingsPagesTest,
+	usersAndOrganizationsPagesTest
+);
+
+test('LPD-46913 language should change properly for admins even if the site does not have the admin language', async ({
+	apiHelpers,
+	page,
+	siteSettingsLocalizationPage,
+	siteSettingsPage,
+	userLocaleOptionsPage,
+}) => {
+	const site = await apiHelpers.headlessSite.createSite({
+		name: getRandomString(),
+	});
+
+	apiHelpers.data.push({id: site.id, type: 'site'});
+
+	await siteSettingsLocalizationPage.goto(site.friendlyUrlPath);
+
+	await siteSettingsLocalizationPage.setCustomDefaultLanguage(
+		'Spanish (Spain)',
+		site.friendlyUrlPath
+	);
+
+	await siteSettingsLocalizationPage.disableAllLanguagesExceptSp(
+		site.friendlyUrlPath
+	);
+
+	const siteURL = `/es/group${site.friendlyUrlPath}`;
+	await page.goto(siteURL);
+
+	await siteSettingsPage.goToSiteSetting(
+		'Localización',
+		'Idiomas',
+		site.friendlyUrlPath
+	);
+
+	await userLocaleOptionsPage.changeLanguageWithAlert();
+
+	expect(
+		siteSettingsLocalizationPage.customDefaultLanguageOption
+	).toBeVisible();
+});
```

Comment:
don't we have to change back the locale to the original one to avoid next tests to fail?

Reply:
1. The test should be changing the language back to the original one (the Admin's language) per the test's requirements.
2. Even if the test fails, the locale is only being set for the site which should be deleted after the test, so I don't believe it should cause any other changes? I might be wrong though, so I can add a finally clause if you feel it's necessary.